File: code/python/numJewelsInStones.py
class Solution:
    def numJewelsInStones(self, J: str, S: str) -> int:
        lst = []
        if J == "":
            return 0
        res = 0
        for i in range(len(J)):
            if J[i] not in lst:
                lst.append(J[i])
        for i in range(len(S)):
            if S[i] in lst:
                res += 1
        return res

File: code/java/numJewelsInStones.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        if(J == "")
            return 0;
        List<Character> lst = new ArrayList<Character>();
        for(int i=0;i<J.length();i++)
            if(!lst.contains(J.charAt(i)))
                lst.add(J.charAt(i));
        int res = 0;
        for(int i=0;i<S.length();i++)
            if(lst.contains(S.charAt(i)))
                res++;
        return res;
    }
}
